frontend/src/metabase/visualizations/lib/y_axis_split.js

```diff
@@ -0,0 +1,114 @@
+import d3 from "d3";
+import _ from "underscore";
+
+import { isStacked } from "./renderer_utils";
+
+export default function getYAxisSplit(series, settings) {
+  const datas = series.map(s => s.data.rows);
```

Comment:
If I'm following correctly, `yExtents` is now being computed twice, using two different methods:

```
const yExtents = datas.map(d => d3.extent(d.map(row => row[1])));
```

```
const yExtents = groups.map(group => d3.extent(group[0].all(), d => d.value));
```

Is there a reason they are different? Could we put `yExtents` into a (hidden) computed setting so it only needs to be calculated once?

Reply:
Actually, IIRC this might be one reason I held off moving the split axis logic into a setting. If those two methods are equivalent this is fine, but I'm not 100% sure they are, since the logic to get `groups` isn't simple.

Reply:
Yeah good catch. I left the one in LAB untouched and didn't have access to `groups` in the settings code so had to recompute it there.

I'd guess the perf concerns of generating extents twice is negligible, but it's bad to have competing definitions. How do you feel about just changing the `yExtents` in LAB to match the one in settings? It feels kinda weird to me to use a hidden setting as a cache, but I'll defer to you on that.

Reply:
Yes at a minimum we should have one implementation of `yExtents` even if we don't cache it in a computed setting. Mostly I'm concerned that your new implementation is not equivalent to the existing one. There's a fair amount of logic that goes into computing `groups` depending on various settings. See all the `getDimensionsAndGroups*` functions.

Reply:
Thanks for pointing me towards the `getDimensionsAndGroups*` functions. Having read that I'm back to having two definitions of `yExtents`, but I _think_ that's ok.

We only auto split axes for charts that are not scatter plots and not stacked, so we just have to match the groups resulting from `getDimensionsAndGroupsForOther`. I think we do that except for any type conversions by loading/dumping from crossfilter (moment objects appear?). As far as I can tell, those shouldn't matter.

I moved the yExtents definition to within the if block that ensures we're dealing with non-scatter, not-stacked charts. Though it wasn't used in the other branch anyways. I also added a comment.

Even if my reasoning is correct and having two definitions of `yExtents` is safe, I realize the logic is pretty convoluted and it will be hard to understand for future readers. Here are two alternatives:

1. Move groups and dimensions into hidden settings. This feel slightly more natural to me than moving yExtents to a setting, but they're roughly equivalent.
2. Avoid having something called `yExtents` in y_axis_split.js. We only compute these so that we can try and match up compatible axes. We could get `yLengths` instead since that's all the `axisCost` function actually looks at. It's effectively the same as what we have now, but would prevent confusion with "what's this subtly different definition of yExtents".